### PR TITLE
Parallel merge of uniqExactIf states

### LINKED ISSUE #49885: Parallelise merging of `UniqExactIf` states
Original question from user: https://github.com/ClickHouse/ClickHouse/discussions/49865#discussioncomment-5904525

We need to repeat what was done for `uniqExact`, i.e. provide [methods](https://github.com/nickitat/ClickHouse/blob/master/src/AggregateFunctions/AggregateFunctionUniq.h#L416-L424) for `AggregateFunctionIf` that is used for parallel merging, that could just call corresponding methods on the nested function.

### PR DESCRIPTION
**Pull request by ucasfl**
### Changelog category (leave one):

- Performance Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Parallel merge of `uniqExactIf` states. Closes #49885.

Before:
```sql
:) select uniqExactIf(number, 1) from numbers_mt(1e8);

SELECT uniqExactIf(number, 1)
FROM numbers_mt(100000000.)

Query id: 59752993-5e0a-45d6-afc9-1711cfde89a8

┌─uniqExactIf(number, 1)─┐
│              100000000 │
└────────────────────────┘

1 row in set. Elapsed: 7.744 sec. Processed 100.00 million rows, 800.00 MB (12.91 million rows/s., 103.31 MB/s.)
```

Now:
```sql
:) select uniqExactIf(number, 1) from numbers_mt(1e8);

SELECT uniqExactIf(number, 1)
FROM numbers_mt(100000000.)

Query id: fcc96bd1-10bd-401b-ba43-4ca20db9edbb

┌─uniqExactIf(number, 1)─┐
│              100000000 │
└────────────────────────┘

1 row in set. Elapsed: 1.291 sec. Processed 100.00 million rows, 800.00 MB (77.45 million rows/s., 619.59 MB/s.)
```

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
This is an automated comment for commit ec82c657eb13b6998deefc94469a8f17c9183c4e with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/50285/ec82c657eb13b6998deefc94469a8f17c9183c4e/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: tests/performance/uniqExactIf.xml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<test>
+    <query>SELECT uniqExactIf(number, ) FROM numbers_mt(1e6)</query>
```
Comment (alexey-milovidov) [inline tests/performance/uniqExactIf.xml:2]: Typo.

**Comment by Algunenano** [comment]
Does it make sense to enable this to other aggregate combinators too?

**Comment by alexey-milovidov** [comment]
@Algunenano Yes.

**Comment by ucasfl** [comment]
> Does it make sense to enable this to other aggregate combinators too?

I will do it in later PR.

**Pull request merged**